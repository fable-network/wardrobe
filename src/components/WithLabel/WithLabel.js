import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Ellipsis = ({ ellipsis, ...otherProps }) => <div {...otherProps} />;
Ellipsis.propTypes = { ellipsis: PropTypes.bool };

const ellipsisCss = `
  overflow: hidden;
  white-space: nowrap;  
  text-overflow: ellipsis;
`;

const Label = styled(Ellipsis)`
  max-width: 100%;
  ${p => p.ellipsis && ellipsisCss};
  font-weight: ${p => p.theme.fontWeightBold};
  color: ${p => p.theme.grey01};
`;

const Content = styled(Ellipsis)`
  ${p => p.ellipsis && ellipsisCss};
  max-width: 100%;
  font-size: ${p => p.theme.fontSizeBase};
  line-height: ${p => p.theme.lineHeightBase};
  color: inherit;
`;

const Wrapper = styled(({ direction, ...otherProps }) => <div {...otherProps} />)`
  display: flex;
  flex-flow: ${p => (p.direction === 'vertical' ? 'column' : 'row')} nowrap;
  align-items: ${p => (p.direction === 'vertical' ? 'flex-start' : 'center')};
  justify-content: flex-start;

  ${p => p.direction === 'horizontal' && '> * { width: 50%; }'};

  font-family: inherit;
  font-weight: ${p => p.theme.fontWeightNormal};
  font-size: ${p => p.theme.fontSizeSmall};
  line-height: ${p => p.theme.lineHeightSmall};
  color: ${p => p.theme.grey02};
`;

const WithLabel = ({ direction, label, children, ...otherProps }) => (
  <Wrapper direction={direction} {...otherProps}>
    <Label>{label}</Label>
    <Content>{children}</Content>
  </Wrapper>
);

WithLabel.defaultProps = {
  direction: 'vertical',
  ellipsis: false,
};

WithLabel.propTypes = {
  label: PropTypes.string.isRequired,
  children: PropTypes.node,
  direction: PropTypes.oneOf(['horizontal', 'vertical']),
  ellipsis: PropTypes.bool,
};

export default WithLabel;
