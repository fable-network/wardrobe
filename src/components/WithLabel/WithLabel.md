WithLabel component to render a given value with a label on top:

```jsx
<WithLabel label="Total WSP">
  EUR <span style={{ textDecoration: 'line-through' }}>15,982.09</span>{' '}
  <span style={{ color: 'red' }}>12,345.67</span>
</WithLabel>
```

With not too much content:

```jsx
<WithLabel label="Pieces">0</WithLabel>
```

Can also be horizontal:

```jsx
<WithLabel label="RRP" direction="horizontal" style={{ width: '300px' }}>
  EUR 69.99
</WithLabel>
```
