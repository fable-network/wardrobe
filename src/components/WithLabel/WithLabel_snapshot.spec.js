/* eslint-disable */
// This file was automatically generated
import React from 'react';
import { shallow } from 'enzyme';
import WithLabel from './WithLabel';

describe('WithLabel Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = shallow(
        <WithLabel label="Total WSP">
          EUR <span style={{ textDecoration: 'line-through' }}>15,982.09</span>{' '}
          <span style={{ color: 'red' }}>12,345.67</span>
        </WithLabel>
      );
      expect(wrapper1).toMatchSnapshot();

      const wrapper2 = shallow(
        <WithLabel label="Pieces">0</WithLabel>
      );
      expect(wrapper2).toMatchSnapshot();

      const wrapper3 = shallow(
        <WithLabel label="RRP" direction="horizontal" style={{ width: '300px' }}>
          EUR 69.99
        </WithLabel>
      );
      expect(wrapper3).toMatchSnapshot();

    });
  });
});
