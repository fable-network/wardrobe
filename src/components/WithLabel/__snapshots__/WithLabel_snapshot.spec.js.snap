// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithLabel Component snapshots should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithLabel
    direction="vertical"
    ellipsis={false}
    label="Total WSP"
  >
    EUR 
    <span
      style={
        Object {
          "textDecoration": "line-through",
        }
      }
    >
      15,982.09
    </span>
     
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      12,345.67
    </span>
  </WithLabel>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithLabel__Label
          ellipsis={false}
        >
          Total WSP
        </WithLabel__Label>,
        <WithLabel__Content
          ellipsis={false}
        >
          EUR 
          <span
            style={
              Object {
                "textDecoration": "line-through",
              }
            }
          >
            15,982.09
          </span>
           
          <span
            style={
              Object {
                "color": "red",
              }
            }
          >
            12,345.67
          </span>
        </WithLabel__Content>,
      ],
      "direction": "vertical",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Total WSP",
          "ellipsis": false,
        },
        "ref": null,
        "rendered": "Total WSP",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "EUR ",
            <span
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              15,982.09
            </span>,
            " ",
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
              12,345.67
            </span>,
          ],
          "ellipsis": false,
        },
        "ref": null,
        "rendered": Array [
          "EUR ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "15,982.09",
              "style": Object {
                "textDecoration": "line-through",
              },
            },
            "ref": null,
            "rendered": "15,982.09",
            "type": "span",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "12,345.67",
              "style": Object {
                "color": "red",
              },
            },
            "ref": null,
            "rendered": "12,345.67",
            "type": "span",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithLabel__Label
            ellipsis={false}
          >
            Total WSP
          </WithLabel__Label>,
          <WithLabel__Content
            ellipsis={false}
          >
            EUR 
            <span
              style={
                Object {
                  "textDecoration": "line-through",
                }
              }
            >
              15,982.09
            </span>
             
            <span
              style={
                Object {
                  "color": "red",
                }
              }
            >
              12,345.67
            </span>
          </WithLabel__Content>,
        ],
        "direction": "vertical",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Total WSP",
            "ellipsis": false,
          },
          "ref": null,
          "rendered": "Total WSP",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "EUR ",
              <span
                style={
                  Object {
                    "textDecoration": "line-through",
                  }
                }
              >
                15,982.09
              </span>,
              " ",
              <span
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                12,345.67
              </span>,
            ],
            "ellipsis": false,
          },
          "ref": null,
          "rendered": Array [
            "EUR ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "15,982.09",
                "style": Object {
                  "textDecoration": "line-through",
                },
              },
              "ref": null,
              "rendered": "15,982.09",
              "type": "span",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "12,345.67",
                "style": Object {
                  "color": "red",
                },
              },
              "ref": null,
              "rendered": "12,345.67",
              "type": "span",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`WithLabel Component snapshots should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithLabel
    direction="vertical"
    ellipsis={false}
    label="Pieces"
  >
    0
  </WithLabel>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithLabel__Label
          ellipsis={false}
        >
          Pieces
        </WithLabel__Label>,
        <WithLabel__Content
          ellipsis={false}
        >
          0
        </WithLabel__Content>,
      ],
      "direction": "vertical",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Pieces",
          "ellipsis": false,
        },
        "ref": null,
        "rendered": "Pieces",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "0",
          "ellipsis": false,
        },
        "ref": null,
        "rendered": "0",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithLabel__Label
            ellipsis={false}
          >
            Pieces
          </WithLabel__Label>,
          <WithLabel__Content
            ellipsis={false}
          >
            0
          </WithLabel__Content>,
        ],
        "direction": "vertical",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Pieces",
            "ellipsis": false,
          },
          "ref": null,
          "rendered": "Pieces",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "0",
            "ellipsis": false,
          },
          "ref": null,
          "rendered": "0",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`WithLabel Component snapshots should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithLabel
    direction="horizontal"
    ellipsis={false}
    label="RRP"
    style={
      Object {
        "width": "300px",
      }
    }
  >
    EUR 69.99
  </WithLabel>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithLabel__Label
          ellipsis={false}
        >
          RRP
        </WithLabel__Label>,
        <WithLabel__Content
          ellipsis={false}
        >
          EUR 69.99
        </WithLabel__Content>,
      ],
      "direction": "horizontal",
      "style": Object {
        "width": "300px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "RRP",
          "ellipsis": false,
        },
        "ref": null,
        "rendered": "RRP",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "EUR 69.99",
          "ellipsis": false,
        },
        "ref": null,
        "rendered": "EUR 69.99",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithLabel__Label
            ellipsis={false}
          >
            RRP
          </WithLabel__Label>,
          <WithLabel__Content
            ellipsis={false}
          >
            EUR 69.99
          </WithLabel__Content>,
        ],
        "direction": "horizontal",
        "style": Object {
          "width": "300px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "RRP",
            "ellipsis": false,
          },
          "ref": null,
          "rendered": "RRP",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "EUR 69.99",
            "ellipsis": false,
          },
          "ref": null,
          "rendered": "EUR 69.99",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
